tests/testsuite/update.rs

```diff
@@ -1646,21 +1646,124 @@ fn update_breaking_unstable() {
         .file(
             "Cargo.toml",
             r#"
+            [package]
+            name  =  "foo"
+            version  =  "0.0.1"
+            edition  =  "2015"
+            authors  =  []
+        "#,
+        )
+        .file("src/lib.rs", "")
+        .build();
+
+    p.cargo("update --breaking")
+        .masquerade_as_nightly_cargo(&["update-breaking"])
+        .with_status(101)
+        .with_stderr_unordered(
+            "\
+[ERROR] the `--breaking` flag is unstable, pass `-Z unstable-options` to enable it
+See https://github.com/rust-lang/cargo/issues/12425 for more information about the `--breaking` flag.
+",
+        )
+        .run();
+}
+
+#[cargo_test]
+fn update_breaking_dry_run() {
+    Package::new("incompatible", "1.0.0").publish();
+    Package::new("ws", "1.0.0").publish();
+
+    let p = project()
+        .file(
+            "Cargo.toml",
+            r#"
+                # Check if formatting is preserved. Nothing here should change, due to dry-run.
+
+                [workspace]
+                members  =  ["foo"]
+
+                [workspace.dependencies]
+                ws  =  "1.0"  # Preserve formatting
+            "#,
+        )
+        .file(
+            "foo/Cargo.toml",
+            r#"
+                # Check if formatting is preserved. Nothing here should change, due to dry-run.
+
                 [package]
                 name  =  "foo"
                 version  =  "0.0.1"
                 edition  =  "2015"
                 authors  =  []
+
+                [dependencies]
+                incompatible  =  "1.0"  # Preserve formatting
+                ws.workspace  =  true  # Preserve formatting
             "#,
         )
-        .file("src/lib.rs", "")
+        .file("foo/src/lib.rs", "")
         .build();
 
-    p.cargo("update --breaking")
+    p.cargo("generate-lockfile").run();
+
+    Package::new("incompatible", "1.0.1").publish();
+    Package::new("ws", "1.0.1").publish();
+
+    Package::new("incompatible", "2.0.0").publish();
+    Package::new("ws", "2.0.0").publish();
+
+    p.cargo("update -Zunstable-options --dry-run --breaking")
         .masquerade_as_nightly_cargo(&["update-breaking"])
-        .with_status(1)
-        .with_stderr_contains("[ERROR] unexpected argument '--breaking' found")
+        .with_stderr(
+            "\
+[UPDATING] `dummy-registry` index
+[UPGRADING] incompatible ^1.0 -> v2.0.0
+[UPGRADING] ws ^1.0 -> v2.0.0
+[LOCKING] 2 packages to latest compatible versions
+[UPDATING] incompatible v1.0.0 -> v2.0.0
+[UPDATING] ws v1.0.0 -> v2.0.0
+[WARNING] not updating lockfile due to dry run
+[WARNING] not upgrading manifest `[..]/Cargo.toml` due to dry run
+[WARNING] not upgrading manifest `[..]/foo/Cargo.toml` due to dry run
```

Comment:
I feel like we don't need three dry-run warnings.

- It'd be messy to implement but we could say "not writing the following files due to dry run" and list them
- We could say "not writing updates due to dry run"
- We could be even more generic and say "aborting update due to dry-run" like some other commands do

Reply:
Something feels off to me about that.  Overly casual?  A blend of detail (files) but vagueness (which ones?)?

Our existing messages:
```console
$ cargo update -n
warn: not updating lockfile due to dry run
$ cargo clean -n
warn: no files deleted due to --dry-run
$ cargo publish -n
warn: aborting upload due to dry run
$ cargo rm foo -n
warn: aborting remove due to dry run
$ cargo add foo -n
warn: aborting add due to dry run
```
(`cargo add` and `cargo rm` were modeled off of `cargo publish` without any deeper thought)

Reply:
OK, I changed this to "aborting update due to dry run".